### ISSUE TITLE
Module hashivault_list seems broken

### ISSUE THREAD

**Issue by lperrin-obs** [TerryHowe/ansible-modules-hashivault#188]
I recently upgraded to last module version (4.2.3)
I use the hashivault_list but since 4.2.0 version, it seems broken. The list returned by the module is empty.

I rollback to 4.1.0 and with the same playbook, the module returns a non-empty list. 

Playbook:
```
- hosts: localhost
  tasks:
  - name: check if password is already created
    hashivault_list:
      secret: "/team/clouds/rp-apache/cloud-rp-z01n-prd.cloud.620nm.net/EID00001_demo"
    register: vault_list
    tags: apache,rp,vhost,password,test

  - debug:
      var: vault_list
    tags: test
```

Result with 4.2.3:
```
PLAY [localhost] ***************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************
Thursday 16 January 2020  09:37:42 +0100 (0:00:00.086)       0:00:00.086 ****** 
ok: [localhost]

TASK [check if password is already created] ************************************************************************************************
Thursday 16 January 2020  09:37:43 +0100 (0:00:01.216)       0:00:01.302 ****** 
ok: [localhost]

TASK [debug] *******************************************************************************************************************************
Thursday 16 January 2020  09:37:44 +0100 (0:00:00.713)       0:00:02.015 ****** 
ok: [localhost] => {
    "vault_list": {
        "changed": false, 
        "failed": false, 
        "rc": 0, 
        "secrets": []
    }
}

PLAY RECAP *********************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Result with 4.1.0:
```
PLAY [localhost] ***************************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************************
Thursday 16 January 2020  09:37:31 +0100 (0:00:00.115)       0:00:00.115 ****** 
ok: [localhost]

TASK [check if password is already created] ************************************************************************************************
Thursday 16 January 2020  09:37:32 +0100 (0:00:01.339)       0:00:01.454 ****** 
ok: [localhost]

TASK [debug] *******************************************************************************************************************************
Thursday 16 January 2020  09:37:33 +0100 (0:00:00.699)       0:00:02.154 ****** 
ok: [localhost] => {
    "vault_list": {
        "changed": false, 
        "failed": false, 
        "rc": 0, 
        "secrets": [
            "rp_apache"
        ]
    }
}

PLAY RECAP *********************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

**Comment by nikokio** [comment]
Hello,

Same here, in order to not downgrade the module, I had to replace `hashivault_list` by an api call using the `uri` module (ansible 2.8 mini required in order to support the LIST method) :

```
- hosts: localhost
  tasks:
  - name: check if password is already created
    uri:
      url: "/team/clouds/rp-apache/cloud-rp-z01n-prd.cloud.620nm.net/EID00001_demo"
      method: LIST
      return_content: yes 
      body_format: json
      status_code: 200 
      headers:
        X-Vault-Token: "{{ vault_token }}"
    register: vault_list
    tags: apache,rp,vhost,password,test

  - debug:
      var: vault_list
    tags: test
```

**Comment by goozbach** [comment]
I am running into the same issue.  I'm seeing the request and response from the Vault side:
```
{
  "time": "2020-02-26T00:18:58.883777311Z",
  "type": "response",
  "auth": {
    "client_token": "hmac-sha256:bc64d4b341b37db00aa1e32b7f8f66563ad0e07dcd8be881fc99e7139b1c1e0e",
    "accessor": "hmac-sha256:ec8920fc96498eb96502d0ae0bf31fc62733fdb59934818f08b57d01ec53b527",
    "display_name": "user@example.com",
    "policies": [
      "default",
      "vault-admins"
    ],
    "token_policies": [
      "default",
      "vault-admins"
    ],
    "metadata": {
      "policies": "vault-admins,vault-admins",
      "username": "user@example.com"
    },
    "entity_id": "fd60a4af-1f06-ed01-f862-67579b6be5cd",
    "token_type": "service"
  },
  "request": {
    "id": "28684415-e998-91a5-4e93-fa2ce1b4022c",
    "operation": "list",
    "client_token": "hmac-sha256:bc64d4b341b37db00aa1e32b7f8f66563ad0e07dcd8be881fc99e7139b1c1e0e",
    "client_token_accessor": "hmac-sha256:ec8920fc96498eb96502d0ae0bf31fc62733fdb59934818f08b57d01ec53b527",
    "namespace": {
      "id": "root"
    },
    "path": "secret/operations/database-master/",
    "remote_address": "10.9.9.21"
  },
  "response": {
    "data": {
      "keys": [
        "hmac-sha256:01190d21f0ab1bae52c188fe9b3c2f872db198d08ea85c6aa76c40daacb1a00f",
        "hmac-sha256:c4ba89fd06988365145ea8bee11d178694e8f2caa84b639726f6eb594394a48c",
        "hmac-sha256:b813553534368e44ef5ea7d872bdc5da378de9a600a2f0a1cc47fb9bd00cf068",
        "hmac-sha256:8d1456fd230fe9e3ddf4b3b63f7ae3edc1ae4cd9c38b701527a6ee3691c05e94",
        "hmac-sha256:c4299d5df09f595b54e1bef82add90f33d9bd850457071de75f447317675014f",
        "hmac-sha256:62be479268f1775ea3247fabb6c3ebb446cb73bfbaf6c548181994ca3b388e36",
        "hmac-sha256:9bd170aefe6e51e66966824b86dc53f787765ef435b4cc466844cb94b3639b1b",
        "hmac-sha256:892499d2df3d105169446330165540f6719b12519958082fcdda3b047053f90d",
        "hmac-sha256:5a10fe0e3fd6bf3502359547980ad3c3219713579dea94de1d8dd545d25b8d5e",
        "hmac-sha256:a3e874546497571ad160f9dde27cbd34e946ca6c7061ec4b2d629bb738768a25",
        "hmac-sha256:c03e7048c914be87026335ccc51e464ab5543a90973377901350c4ac020be3fa",
        "hmac-sha256:e1f38755b0d1638bb7a673fc078e9606f2f4641b3cdf708c54c2ae6a45a11f74",
        "hmac-sha256:a4993ee07ed7814c5a99330a707932157582b54c8c649d090d0695048cd1b352",
        "hmac-sha256:69442cb8517f27285939fe2d39f234db4fa9035b456ff264b938a08d901bff94",
        "hmac-sha256:4fa0da74cb18efb26eb7fe32029f6303d022694e4f463ff4766e66d3bf3953d2",
        "hmac-sha256:dbbee4468a714be82edc2a6dca3722837fed14631033c19c94316917a26c09bd",
        "hmac-sha256:3d34c44e873fb3744661fe231556514d49148edd7c222b0ce5b25351b94b3892",
        "hmac-sha256:d17011e5c3c31b531dcaa2f4f81593ed69bcbc848c338d767f8c54a10bd9a218",
        "hmac-sha256:4d2c818d6165b6848fe974c7b6ee5ad7a718c75ee53d60af546d0908820d252d",
        "hmac-sha256:08450d01d7b37b9674365843b5a5b5783180d8b3016eebd96fc3867816ad2821",
        "hmac-sha256:c9cac1094da8793d234cc1816f54b21318ef3f23c21b2ed71a26d3cc06febf0a",
        "hmac-sha256:1a6933ccebad3430c58452f4464cfc19b3cda64e7c3af185f0b739705870ab55",
        "hmac-sha256:3763944f56bf372874938b13c7943817d3a74efb34492b333a2f62bd311e7517"
      ]
    }
  }
}
```
I've verified that the contents hash out to what they should be.  So it's likely in the part of the code that moves the results from hvac client to results dict.

**Comment by TerryHowe** [comment]
Must be related to the v2 support https://github.com/TerryHowe/ansible-modules-hashivault/commit/d82f701b7a0c7539e71502432322198402d7450e#diff-30b9ab772ef509e132536a7db4f8eb2a 4.2.0 went out in late October

**Comment by TerryHowe** [comment]
We have plenty of tests for relative paths, but none for a full path like you are using.

**Comment by TerryHowe** [comment]
I have something that will provide some backward compatibility, but only for a mount point that is one level. Any other mount_point will require use of the new parameter, we can't guess your mount_point, so best I can do.

**Comment by TerryHowe** [comment]
https://github.com/TerryHowe/ansible-modules-hashivault/commit/915310f294d51473a0ee7e6961b8eb93768598f0#diff-30b9ab772ef509e132536a7db4f8eb2a

**Comment by TerryHowe** [comment]
Should be fixed in 4.3.2
